### ISSUE TITLE
Add parameters for converting to uppercase, lowercase or capitalising of text matching function

### ISSUE THREAD

**Issue by RealRaven2000** [RealRaven2000/SmartTemplates#288]
Text matching functions such as `matchTextFromBody()` will be even more useful if they allow conversion of case - these can be supplied with an additional parameter such as:

```
%matchTextFromBody("Customer.*\n\-+\n(.*)\n",1,capitalize)%
%matchTextFromSubject("Customer.*\n\-+\n(.*)\n",1,uppercase)%
%matchTextFromBody("Customer.*\n\-+\n(.*)\n",1,lowercase)%

%header.prefix.matchFromBody(subject,"Customer.*\n\-+\n(.*)\n",1,capitalize)%
```

**Comment by RealRaven2000** [comment]
I decided to also add support applying these text transformation to the address variables, as in the following example:

```
%recipient(firstname,uppercase,lastname,capitalize)%
%from(name,lowercase)%
```

The switch will format all previous components in the line (but not overwrite earlier transformations of a different switch), so e.g. in the example "1,uppercase,2,3,lowercase"  1 will be in uppercase, 2 and 3 will be in lowercase.


[smartTemplate-fx-4.5pre70.zip](https://github.com/RealRaven2000/SmartTemplates/files/15178967/smartTemplate-fx-4.5pre70.zip)

-----

To install, download archive and drop into Thunderbird Add-ons Manager (don't extract contents).

**Comment by RealRaven2000** [comment]
Implemented in 4.5 released on 14/05/2024
